src/net_processing.h

```diff
@@ -71,12 +71,8 @@ class PeerManager : public CValidationInterface, public NetEventsInterface
     /** Set the best height */
     virtual void SetBestHeight(int height) = 0;
 
-    /**
-     * Increment peer's misbehavior score. If the new value >= DISCOURAGEMENT_THRESHOLD, mark the node
-     * to be discouraged, meaning the peer might be disconnected and added to the discouragement filter.
-     * Public for unit testing.
-     */
-    virtual void Misbehaving(const NodeId pnode, const int howmuch, const std::string& message) = 0;
+    /* Public for unit testing. */
+    virtual void UnitTestMisbehaving(NodeId peer_id, int howmuch) = 0;
```

Comment:
Having a function in the API for unit testing is a bit strange to me, as it's also there when not testing.  I understand the whitebox testing idea, and don't immediately know a better way of doing it, but it's a pattern we imo should try to avoid

Reply:
I don't think there is any other way with the pimpl approach. Though, whenever I see a unit-test-only method in the "normal" interface, I like to prefix it with `UnitTest...` to make it absolutely obvious when someone is calling this from normal code.

It may be possible to remove by sending a mock violation, but I can look into this as a follow-up. (This pull doesn't introduce the method, just renames it).

Reply:
Yes, fair enough. Agree with the naming convention, at least it discourages use in normal code. With LTO, it may even completely get optimized out.